0004/README.md

```diff
@@ -0,0 +1,416 @@
+    1. (Optional) Build ID
+    1. (Optional) List of URIs that resolve to proof of SLSA conformance
+    1. (Optional) List of URIs that resolve to materials for verifying
+        provenance
+    1. (Optional) List of URIs that resolve to the build platform's
+        public vulnerability disclosures
+
+1. Complete the terms and conditions form.
+1. Send the PR for review.
+1. Have the auditor approve the PR with a comment indicating the audited
+    SLSA levels
+
+The maintainers will review the PR and merge it after the auditor's approval.
+
+Once the PR is merged and the terms are signed, the OpenSSF grants the build
+platform a limited license to use the SLSA logo on its website and other
+promotional materials. Should the build platform fail to adhere to the program's
+terms or meet its requirements, the OpenSSF will revoke that license and, if
+necessary, enforce its trademarks. 
+
+### Auditor participation
+
+Auditors can take part in the program by:
+
+1. Registering their intent with the OpenSSF.
+1. Performing any training and/or certification required by the OpenSSF (TBD)
+1. Completing the terms and conditions form.
+1. Opening a PR against the auditor repository containing 
+    1. Name of the auditor
+    1. Auditor's website
+    1. A URI that resolves to the auditor's auditing procedure. 
+
+The community will review the PR and merge it once it has been approved by a
+quorum of maintainers. It is not yet clear what, if any, criteria will be used
+to evaluate auditors.   
+
+Once the PR is merged and the terms and signed, the OpenSSF grants the auditor a
+license to use the SLSA logo on its website and promotional materials. The
+OpenSSF reserves the right to revoke the license if the auditor fails to
+publicize audit reports,  performs insufficiently thorough audits, or otherwise
+acts against the SLSA Community's interests.  
+
+## Alternatives considered
+
+### Central repository
+
+The OpenSSF would maintain a repository that stored all public proof of
+conformance and provenance verification materials for build platforms taking
+part in the program. This option made enforcement easy since you could remove
+build platforms that do not comply with the program's terms, but it puts too
+much trust in the repository's maintainers. This option would also require
+infrastructure for build platforms to rotate and revoke provenance verification
+materials, which would complicate the design.
+
+### Build platform attestations
+
+Rather than maintain offline proof of conformance that would be written and
+interpreted by humans, this option would capture the state of the build platform
+in an attestation and attach it to all provenance produced by the build
+platform. We discarded this option because it would violate SLSA's guiding
+principle of trusting systems by moving conformance checks into provenance
+verification. It would also bloat the provenance size and require introducing
```

Comment:
I disagree with these particular reservations, because:
1. We shouldn't conflate the build _provenance_ with any new build _platform_ attestation.
2. New in-toto predicates will come in the future, thus necessarily increasing the total size of any bundle of attestations.
3. (2) will necessarily add new logic to any SLSA verifier.

Reply:
I think we're talking about different ideas. I was thinking about attestations about the machine state that would be validated against a SLSA conformance policy or test suite, but you were suggesting attestations for auditors to attest to SLSA levels.

I'm against attestations about machine state because I don't think it's possible to write a policy or test that describes a sufficiently secure build platform. I'm aware of some efforts to do so, but I don't think they're widely accepted enough to be part of SLSA.

I don't have strong feelings about auditor attestations in general. However, I don't think they're necessary to start a conformance program because we could easily add them later on top of the registry. We would also need to engage with the complexity of how to use those attestations (e.g. at each verification, once when configuring the verifier's root of trust?), and I would prefer to start with a simple approach and add complexity as needed.